packages/core/src/components/Link/Link.tsx

```diff
@@ -35,7 +37,7 @@ export type LinkProps = MaterialLinkProps &
  */
 export const Link = React.forwardRef<any, LinkProps>((props, ref) => {
   const to = String(props.to);
-  return /^https?:\/\//.test(to) ? (
+  return isExternalUri(to) ? (
```

Comment:
Wait, wait. Let me revise my review on this. Stepping back a bit to think what it is that we are doing.

We should not necessarily ask ourselves "is this external or not". We should be asking "can react-router handle this or not". And THAT test could probably be "starts with dot or slash".

Everything else - whether it's valid or invalid, whether it has a protocol at the start or not, etc - should just be handled by the regular link.

To provide some context, our internal Link has even more rules. You can do things like "if the protocol is https and the hostname is our-full-hostname, then take the path part and use react-router to navigate to that". Even though the input happened to actually formally be an external URL.

Reply:
So are you thinking that the check should literally be `to.startsWith('/')` (or a simplified regex) to indicate it's not external (i.e. an `href`)? I'm actually not sure what in what cases routes would start with `.` are though?

Reply:
Oh pff, as @Rugvip correctly noted, even the string `https` (starting with neither slash nor dot) should be handled just as `./https` would. This is tricky :) It's almost like if it contains a colon at all, then use the browser, and otherwise use react-router

Reply:
Well there's also the `//some-host/page` syntax as well which uses the current scheme of the host location 😞.